src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -229,47 +231,61 @@ await foreach (ConfigurationSetting setting in _client.GetConfigurationSettingsA
                         LabelFilter = loadOption.LabelFilter
                     };
 
-                    // Load all key-values with the null label.
                     await CallWithRequestTracing(async () =>
                     {
                         await foreach (ConfigurationSetting setting in _client.GetConfigurationSettingsAsync(selector, CancellationToken.None).ConfigureAwait(false))
                         {
-                            data[setting.Key] = setting;
+                            serverData[setting.Key] = setting;
                         }
                     }).ConfigureAwait(false);
                 }
 
                 // Block current thread for the initial load of key-values registered for refresh that are not already loaded
-                await Task.Run(() => LoadKeyValuesRegisteredForRefresh(data).ConfigureAwait(false).GetAwaiter().GetResult()).ConfigureAwait(false);
+                await Task.Run(() => LoadKeyValuesRegisteredForRefresh(serverData).ConfigureAwait(false).GetAwaiter().GetResult()).ConfigureAwait(false);
+
+                if (_options.OfflineCache != null)
+                {
+                    _options.OfflineCache.Export(_options, JsonSerializer.Serialize(serverData));
+                }
+
+                data = serverData;
             }
             catch (Exception exception) when (exception is RequestFailedException ||
                                               ((exception as AggregateException)?.InnerExceptions?.All(e => e is RequestFailedException) ?? false) ||
                                               exception is OperationCanceledException)
             {
                 if (_options.OfflineCache != null)
                 {
-                    data = JsonSerializer.Deserialize<IDictionary<string, ConfigurationSetting>>(_options.OfflineCache.Import(_options));
-
-                    if (data != null)
+                    // During startup or refreshAll scenario, we'll try to populate config from offline cache, if available
+                    string cachedData = _options.OfflineCache.Import(_options);
+                    if (cachedData != null)

```

Comment:
>if (cachedData != null) [](start = 20, length = 23)

I would suggest leaving a blank line before each block statement (such as if, foreach etc.) unless it is the first statement in the method. This will improve readability and maintain consistency with existing code.